src/webgpu/api/validation/buffer/mapping.spec.ts

```diff
@@ -369,6 +389,61 @@ Test for various cases of being unmapped: at creation, after a mapAsync call or
     }
   });
 
+g.test('getMappedRange,subrange,mapped')
+  .desc(
+    `Test that old getMappedRange returned arrybuffer does not exist after unmap and newly returned
+    arraybuffer after new map has correct subrange`
+  )
+  .params(u => u.combine('mapMode', kMapModeOptions))
+  .fn(async t => {
+    const { mapMode } = t.params;
+    const bufferSize = 16;
+    const offset = 8;
+    const subrangeSize = bufferSize - offset;
+    const buffer = t.createMappableBuffer(mapMode, bufferSize);
+    await buffer.mapAsync(mapMode);
+
+    const data0 = buffer.getMappedRange();
+    t.expect(data0 instanceof ArrayBuffer);
+    t.expect(data0.byteLength === bufferSize);
+
+    buffer.unmap();
```

Comment:
nit: can do one extra check (I'm sure we already test this elsewhere)
```suggestion
    buffer.unmap();
    t.expect(data0.byteLength === 0);
```